scripts/generate-upgrade-dashboard.js

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+import semver from 'semver';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+// get latest stable Backstage release
+async function getLatestBackstageVersion() {
+  const response = await fetch(
+    'https://api.github.com/repos/backstage/backstage/releases',
+  );
+  if (!response.ok) throw new Error(`HTTP error! status: ${response.status}`);
+
+  const releases = await response.json();
+  const stableRelease = releases.find(
+    release =>
+      !release.prerelease &&
+      !release.tag_name.includes('next') &&
+      !release.tag_name.includes('rc'),
+  );
+
+  if (!stableRelease) throw new Error('No stable releases found');
+  return stableRelease.tag_name;
+}
+
+// get all workspace versions
+function getWorkspaceVersions() {
```

Comment:
Could this be streamlined by using - https://github.com/backstage/community-plugins/blob/main/scripts/list-workspaces.js - have a feeling maybe not now that I see you include the needed Backstage version details but figured I would mention it.

Reply:
I've done this in 6d31699, it does actually make it look a bit cleaner/simpler too. Thanks!